src/components/EditRecipe.tsx

```diff
@@ -0,0 +1,52 @@
+import axios from "axios";
+import React, {useEffect, useState} from "react";
+import * as Router from 'react-router-dom';
+import RecipeForm from "./RecipeForm";
+import Recipe from "./recipe";
+import {ErrorMessage} from "./styles";
+
+const {useNavigate, useParams} = Router;
+
+export default function EditRecipe() {
```

Comment:
another option could have been using the same component for Edition and Creation

Reply:
They both share the RecipeForm component. I created the specific ones for Edition and Creation to use the correct endpoints. I guess that having only one component we could call one endpoint or the other depending on the params.id attribute value